modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml

```diff
@@ -1483,11 +1483,14 @@ components:
     Animal:
       type: object
       discriminator:
-        propertyName: className
+        propertyName: type
+        mapping:
+          DOG: '#/components/schemas/Dog'
+          CAT: '#/components/schemas/Cat'
       required:
-        - className
+        - type
       properties:
-        className:
+        type:
```

Comment:
nit: I would not use `type` here since it's the name of another property of the YAML - maybe `animalType` ?

Reply:
agree, it's confusing. I've changed it to `species`.